src/Security/Authentication/Negotiate/src/Internal/NegotiateState.cs

```diff
@@ -2,7 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Net.Security;
-using System.Security.Authentication;
```

Comment:
Interesting. Build failed complaining about this, but I didn't put together it's because of a `AuthenticationException` name conflict.

I specifically avoided the existing exception because I didn't feel its [description](https://learn.microsoft.com/en-us/dotnet/api/system.security.authentication.authenticationexception?view=net-7.0) aligned with my usage:
> The exception that is thrown when authentication fails for an authentication stream.

Two options:

1. Just use `System.Security.Authentication.AuthenticationException` and drop the new one.
2. Keep this as a breaking change.

Reply:
Kestrel does seem to have protocol-level uses of `System.Security.Authentication.AuthenticationException`:
- https://github.com/dotnet/aspnetcore/blob/ae54fe016e11fc88ed8e4a57f904a35d60cc80be/src/Servers/Kestrel/Core/src/Internal/SniOptionsSelector.cs#L137-L145
- https://github.com/dotnet/aspnetcore/blob/ae54fe016e11fc88ed8e4a57f904a35d60cc80be/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs#L187-L195

Reply:
Don't change kestrel. Change the new exception name not to conflict. `AuthenticationFailureException`?

Reply:
> Don't change kestrel.

:+1: just trying to highlight a semantically distinct use of the existing `AuthenticationException`.

> Change the new exception name not to conflict. `AuthenticationFailureException`?

This works for me, but I'll wait for API review.

Am I interpreting correctly that you're on board with throwing the new exception here?

Reply:
> Am I interpreting correctly that you're on board with throwing the new exception here?

I'd like to find an existing exception type that makes sense here, but I haven't been able to.

Reply:
This is now throwing `AuthenticationFailureException`.